tools/webpack-plugin/src/BacktracePlugin.ts

```diff
@@ -1,19 +1,77 @@
-import webpack from 'webpack';
-import { BacktracePluginV4 } from './BacktracePluginV4';
-import { BacktracePluginV5 } from './BacktracePluginV5';
-
-let BacktracePlugin: typeof BacktracePluginV4 | typeof BacktracePluginV5;
-
-const version = process.env.WEBPACK_VERSION ?? webpack.version[0];
-switch (version) {
-    case '4':
-        BacktracePlugin = BacktracePluginV4;
-        break;
-    case '5':
-        BacktracePlugin = BacktracePluginV5;
-        break;
-    default:
-        throw new Error(`Webpack version ${version} is not supported.`);
-}
+import { DebugIdGenerator, SourceMapUploader, SourceProcessor } from '@backtrace/sourcemap-tools';
+import path from 'path';
+import webpack, { WebpackPluginInstance } from 'webpack';
+import { BacktracePluginOptions } from './models/BacktracePluginOptions';
+
+export class BacktracePlugin implements WebpackPluginInstance {
+    private readonly _sourceMapProcessor: SourceProcessor;
+    private readonly _sourceMapUploader?: SourceMapUploader;
+
+    constructor(public readonly options?: BacktracePluginOptions) {
+        this._sourceMapProcessor = new SourceProcessor(options?.debugIdGenerator ?? new DebugIdGenerator());
+
+        this._sourceMapUploader =
```

Comment:
I didn't get answer to my question about this in the previous diff. How the source map uploader can be undefined? What is the purpose of having our plugin without source map uploader?

Reply:
Well, you could always upload the files manually, if you want. The debug IDs will still be injected.